why3/src/mlcfg/printer.rs

```diff
@@ -1030,7 +1031,20 @@ impl Print for Constant {
             }
             Constant::String(s) => alloc.text(format!("{s:?}")),
             Constant::Uint(i, None) => alloc.as_string(i),
-            Constant::Float(f) => alloc.text(format!("{f:.64}")),
+            Constant::Float(f) => {
+                assert!(!f.is_nan() && f.is_finite() && f.is_normal());
```

Comment:
I understand why you are assuming that f is finite (and non-Nan, which is a corolarry), but there is really no reason it should be normal.

Reply:
The reason I need to is that the printing is false if that isn't the case, and again, is it even possible to write a non-normal float _literal_?

Reply:
You need to do a special case (I know, this is tedious). If the float is subnormal, then multiply it by 2^60, print the mantissa as you do, and shift the exponent by 60.